application/models/Question.php

```diff
@@ -1565,20 +1563,11 @@ public function questionThemeNameValidator()
      * Returns the QuestionTheme related to this question.
      * It's not implemented as a relation because relations only work on
      * persisted models.
-     * It also returns the proper theme when 'question_theme_name' is empty or 'core'.
      *
      * @return QuestionTheme|null
      */
     public function getQuestionTheme()
     {
-        $questionThemeName = $this->question_theme_name;
-        // If the model's question_theme_name attribute is empty or 'core', we get the name for the base theme
-        if (empty($questionThemeName) || $questionThemeName == 'core') {
-            $questionThemeName = QuestionTheme::model()->getBaseThemeNameForQuestionType($this->type);
-        }
-        if (empty($this->relatedQuestionTheme) || $this->relatedQuestionTheme->name != $questionThemeName) {
-            $this->relatedQuestionTheme = QuestionTheme::model()->findByAttributes(['question_type' => $this->type, 'name' => $questionThemeName]);
-        }
-        return $this->relatedQuestionTheme;
+        return $this->getRelated("question_theme", $this->isNewRecord);
```

Comment:
Would it make sense to have second argument always false? That is, always fetch the relation, but throw an exception/return null if object did not set question_theme_name and question_type yet?

Reply:
Hmm not sure... need to think about it.
Why? In which cases do you see an advantage?

Reply:
I was thinking we can use $question->getQuestionTheme() in the views even when it's not saved yet.

Reply:
Maybe add a force_refresh argument?
- if force_reresh, always refresh
- if not forece_refresh, use $this->isNewRecord, as it is now

Reply:
Why? I'm assuming we always know both question theme and question type when entering the view files, both when question is saved and not saved.

Reply:
I say adding an argument to the method as to not always refresh.
Avoid unnecesary refreshes, just that. I am missing something?

Reply:
Ah, ok. Yeah, try it. :+1:

Reply:
> I was thinking we can use $question->getQuestionTheme() in the views even when it's not saved yet. So the logic in the view files always look the same.

FYI, the method is being used as $question->questionTheme. Not sure what you mean by " logic in the view files always look the same", but I believe is already working fine.

The parameter with a default value is harmless still, but wanted to double check before jumpng into it

Reply:
> Not sure what you mean by " logic in the view files always look the same", but I believe is already working fine.

I meant it should look the same for saved and not-yet-saved questions. If that's already the case, great!